flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapKeyedStateBackend.java

```diff
@@ -266,18 +265,20 @@ public void restorePartitionedState(List<KeyGroupsStateHandle> state) throws Exc
 			for (int i = 0; i < numKvStates; ++i) {
 				String stateName = inView.readUTF();
 
-				ObjectInputStream ois = new ObjectInputStream(inView);
+				TypeSerializer namespaceSerializer =
+						InstantiationUtil.deserializeObject(fsDataInputStream, userCodeClassLoader);
+				TypeSerializer stateSerializer =
+						InstantiationUtil.deserializeObject(fsDataInputStream, userCodeClassLoader);
 
-				TypeSerializer namespaceSerializer = (TypeSerializer) ois.readObject();
-				TypeSerializer stateSerializer = (TypeSerializer) ois.readObject();
-				StateTable<K, ?, ?> stateTable = new StateTable(stateSerializer,
+				StateTable<K, ?, ?> stateTable = new StateTable(
+						stateSerializer,
 						namespaceSerializer,
 						keyGroupRange);
 				stateTables.put(stateName, stateTable);
 				kvStatesById.put(i, stateName);
 			}
 
-			for (int keyGroupIndex = keyGroupRange.getStartKeyGroup(); keyGroupIndex <= keyGroupRange.getEndKeyGroup(); keyGroupIndex++) {
+			for (int keyGroupIndex = keyGroupRange.getStartKeyGroup();  keyGroupIndex <= keyGroupRange.getEndKeyGroup(); ++keyGroupIndex) {
```

Comment:
I think i wanted to break this because it exceeds the line limit but then decided against it because IntelliJ messed up the formatting for loops. Nothing wrong there at all.